endpoints/openrtb2/auction_test.go

```diff
@@ -5010,40 +5010,54 @@ func TestParseRequestStoredBidResponses(t *testing.T) {
 			name:             "req imp has valid stored bid response",
 			givenRequestBody: validRequest(t, "imp-with-stored-bid-resp.json"),
 			expectedStoredBidResponses: map[string]map[string]json.RawMessage{
-				"imp-id1": {"testBidder1": bidRespId1},
+				"imp-id1": {"telaria": bidRespId1},
 			},
 			expectedErrorCount: 0,
 		},
 		{
-			name:             "req imp has valid stored bid response with case insensitive bidder name",
-			givenRequestBody: validRequest(t, "imp-with-stored-bid-resp-insensitive-bidder-name.json"),
+			name:             "req imp has valid stored bid response with case not-matching bidder name",
+			givenRequestBody: validRequest(t, "imp-with-stored-bid-resp-case-not-matching-bidder-name.json"),
 			expectedStoredBidResponses: map[string]map[string]json.RawMessage{
-				"imp-id3": {"APPNEXUS": bidRespId3},
+				"imp-id3": {"appnexus": bidRespId3},
+			},
+			expectedErrorCount: 0,
+		},
+		{
+			name:             "req imp has valid stored bid response with case matching bidder name",
+			givenRequestBody: validRequest(t, "imp-with-stored-bid-resp-case-matching-bidder-name.json"),
+			expectedStoredBidResponses: map[string]map[string]json.RawMessage{
+				"imp-id3": {"appnexus": bidRespId3},
 			},
 			expectedErrorCount: 0,
 		},
 		{
 			name:             "req has two imps with valid stored bid responses",
 			givenRequestBody: validRequest(t, "req-two-imps-stored-bid-responses.json"),
 			expectedStoredBidResponses: map[string]map[string]json.RawMessage{
-				"imp-id1": {"testBidder1": bidRespId1},
-				"imp-id2": {"testBidder2": bidRespId2},
+				"imp-id1": {"telaria": bidRespId1},
+				"imp-id2": {"amx": bidRespId2},
 			},
 			expectedErrorCount: 0,
 		},
 		{
 			name:             "req has two imps one with valid stored bid responses and another one without stored bid responses",
 			givenRequestBody: validRequest(t, "req-two-imps-with-and-without-stored-bid-responses.json"),
 			expectedStoredBidResponses: map[string]map[string]json.RawMessage{
-				"imp-id2": {"testBidder2": bidRespId2},
+				"imp-id2": {"amx": bidRespId2},
 			},
 			expectedErrorCount: 0,
 		},
 		{
 			name:                       "req has two imps with missing stored bid responses",
 			givenRequestBody:           validRequest(t, "req-two-imps-missing-stored-bid-response.json"),
 			expectedStoredBidResponses: nil,
-			expectedErrorCount:         2,
+			expectedErrorCount:         0,
```

Comment:
I'm curious what the thought process is here. Why did this behavior change? It looks like both the bidders and the stored response ids in `imp.ext.prebid.storedbidresponse` are invalid in the corresponding JSON file. Is that intended?

The next test case that involves `imp-with-stored-bid-resp-non-existing-bidder-name.json` considers it an error if the bidder is invalid but the stored bid response ID is not.

Reply:
Ok, I see what is happening here. This test executes `deps.parseRequest` where we parse imp.bidders and imp.storedrequests and validate the result.
1) Test with `req-two-imps-missing-stored-bid-response.json` doesn't return errors, because there are no matches here:
```json
 "appnexus": {
            "placementId": 12883451
          },
          "prebid": {
            "storedbidresponse": [
              {"bidder":"testBidder1", "id": "123"}
            ]
          }
```
In this case result stored responses are empty and actual request is going to the "appnexus" adapter. 
We need to confirm if we want this behavior. Before this change if `imp.prebid.storedbidresponse` was specified in imp, then we required stored responses for every bidder in `imp.ext`. Now if there are no matches between `imp.ext.bidder`  and bidders in stored responses - we execute this imp as a "usual" imp and bidders send actual http calls to actual bidders

2) Test with `imp-with-stored-bid-resp-non-existing-bidder-name.json` returnes a validation error for bidder `bidderABC`, because it cannot normalize it. This test has predefined deps with this bidder map:
`map[string]openrtb_ext.BidderName{"telaria": "telaria", "amx": "amx", "appnexus": "appnexus"}`,

Reply:
Gotcha. I can't find any info confirming this but I thought we made a conscious decision a while back to report an error if all bidders called out in imps didn't have a stored bid response when the bid response object is present in the request. I don't recall a specific reason for this implementation but I assume it was to help us avoid confusion in case people think they can mix stored bid responses with real bid requests for an imp, which is something we don't currently support.

I don't think this is critical but I'm having trouble discerning what is driving the change here. Thoughts?

Reply:
Fixed back: if stored responses are specified then every bidder (ignoring casing) in imp.ext should habe a stored bid response